app/src/main/java/ch/epfl/balelecbud/map/MapViewFragment.java

```diff
@@ -29,74 +31,82 @@
 import static ch.epfl.balelecbud.BalelecbudApplication.getAppAuthenticator;
 import static ch.epfl.balelecbud.BalelecbudApplication.getAppDatabase;
 
-public class MapViewActivity extends BasicActivity {
-
-    private final static String TAG = MapViewActivity.class.getSimpleName();
-    private static OnMapReadyCallback mockCallback;
+public class MapViewFragment extends Fragment {
+    public final static String TAG = MapViewFragment.class.getSimpleName();
+    private static com.mapbox.mapboxsdk.maps.OnMapReadyCallback mockCallback;
     private MapView mapView;
-    private MyMap myMap;
+    private static MyMap myMap;
     private Map<User, MyMarker> friendsMarkers = new HashMap<>();
     private List<PointOfInterest> waitingPOI = new LinkedList<>();
     private Map<User, Location> waitingFriendsLocation = new HashMap<>();
     private Location defaultLocation;
 
     @VisibleForTesting
-    public static void setMockCallback(OnMapReadyCallback mockCallback) {
-        MapViewActivity.mockCallback = mockCallback;
+    public static void setMockCallback(com.mapbox.mapboxsdk.maps.OnMapReadyCallback mockCallback) {
+        MapViewFragment.mockCallback = mockCallback;
     }
 
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
+    @VisibleForTesting
+    public static void setMockMap(MyMap map) {
+        myMap = map;
+    }
 
-        Mapbox.getInstance(this, getString(R.string.mapbox_access_token));
+    public static MapViewFragment newInstance() {
+        return new MapViewFragment();
+    }
 
-        setContentView(R.layout.activity_map);
-        mapView = findViewById(R.id.map_view);
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Mapbox.getInstance(getActivity(), getString(R.string.mapbox_access_token));
+        View inflatedView = inflater.inflate(R.layout.fragment_main_map, container, false);
+        mapView = inflatedView.findViewById(R.id.map_view);
         mapView.onCreate(savedInstanceState);
-
-        if (mockCallback != null) {
+        if (myMap != null) {
```

Comment:
Why is this myMap and not mockCallback? It's verified in line 77 that myMap is not null